woocommerce/woocommerce
# Add labels and allow searching terms in WooCommerce Navigation Link block variations

Fixes https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/4130.
Fixes https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/4131.
Fixes https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/4132.

WordPress 5.8 is [adding](https://core.trac.wordpress.org/changeset/50527/trunk/src/wp-includes/post.php) an `item_link` and `item_link_description` label properties to custom post types and taxonomies, so they can be used to add friendly names to Navigation Link block variations. Notice those variations are created automatically from CPT and taxonomies.

In order to get those block variations to work properly, we also need to enable Rest for Product Categories and Product Tags so they can be searched using the UI.

### Screenshots

#### Labels

| Before | After |
| --- | --- |
| ![imatge](https://user-images.githubusercontent.com/3616980/116261022-88066780-a777-11eb-81aa-f40178903c24.png) | ![imatge](https://user-images.githubusercontent.com/3616980/116260763-45448f80-a777-11eb-8b7e-39b7d72fafb5.png)  |

#### Rest suggestions
![imatge](https://user-images.githubusercontent.com/3616980/116260633-2514d080-a777-11eb-9d45-ec51e8be8d54.png)

### How to test the changes in this Pull Request:

1. You will need to install a recent nightly of WordPress 5.8 (you can use [WordPress beta tester](https://wordpress.org/plugins/wordpress-beta-tester/) for this) and the latest version of [Gutenberg](https://wordpress.org/plugins/gutenberg/).
2. Create a page and add a Navigation block.
3. Inside the navigation block, add a product link, a product category link and a product tag link and verify all of them work correctly and have labels that make it clear they are related to products.

### Changelog entry

> Add labels and allow searching terms in WooCommerce Navigation link block variations
